crates/agent/src/evolution.rs

```diff
@@ -232,92 +253,65 @@ async fn process_row(
 }
 
 fn validate_evolving_collections(
-    spec_rows: &Vec<DraftSpecRow>,
-    evolving_collections: &BTreeMap<String, Option<String>>,
+    spec_rows: &Vec<SpecRow>,
+    reqs: &[EvolveRequest],
 ) -> Result<(), String> {
     let collection_name_regex = models::Collection::regex();
     let mut seen = BTreeSet::new();
-    for row in spec_rows
-        .iter()
-        .filter(|r| evolving_collections.contains_key(r.catalog_name.as_str()))
-    {
-        let old_name = row.catalog_name.as_str();
-        seen.insert(old_name.to_owned());
+    for req in reqs {
+        // Make sure there's only one request per collection
+        let old_name = req.current_name.as_str();
```

Comment:
yeah, that's fair, but it seemed preferable to keep all the validations in one place, since we need the `spec_rows` for at least some of the validations. This probably would make sense in conjunction with your other suggestion about consolidating the requests and `spec_rows`, though.